### ISSUE TITLE
Emergency program activated with no apparent reason

### ISSUE THREAD

**Issue by notarocketscientist** [nicymike/serflings#112]
Hi!

I played mission 16 and having enough wooden boards and stones, the "emergency program" is activated.

More detailed description and savegames available here:
https://github.com/notarocketscientist/serflings-debug/tree/master/emergency-program

I read in the manual, that the emergency program activates if you are low on boards and stones _in the castle_. But in my case, I have plenty in a stock building.

My problem is, that the emergency program messes up my new building queue and it takes forever to complete any new/extisting buildings.

**Comment by nicymike** [comment]
Thank you for the issue. I also had this bug some day but I did not get it later when I started to debug the game.
The emergency program is also connected to the three buildings: sawmill, lumberjack and stonecutter. If you build all of this (or at least build it after emergency program gets activated) the program should be deactivated.

I guess there is some error in handling the buildings but I did not find a problem yet.

**Comment by notarocketscientist** [comment]
True, in the [mentioned savegame ](https://github.com/notarocketscientist/serflings-debug/blob/master/emergency-program/README.md#how-to-reproduce) I just deleted the last stonecutter hut. But I have in total 52 wooden boards and 31 stones so this emergency mode should not be triggered in my opinion in the first place.

And, it seems that even when there is a stonecutter built, the emergency mode is not deactivated:

* I have a castle with two stone blocks under construction: this will only be filled up with stones, but no wooden planks (even though the wooden planks from the sawmill pile up in the the nothern stock building)
* new "small" buildings don't get a builder assigned
* new "big" buildings don't even get a guy leveling the foundation

I'll update the bug description with more details. Maybe you can figure this out.

**Comment by notarocketscientist** [comment]
Seems like the issue is not unknown: http://www.mag64.de/index.php?page=nds/siedds/siedds&server=1&ext=1

> Wehe Ihr lest hier die Meldung "Das Notprogramm wurde aktiviert". Je nach Gutdünken des Spiels und damit reine Glückssache, kann es nämlich passieren das trotz kurz darauf fertigem Sägewerk das Notprogramm nicht deaktiviert wird und alle neuen Bretter direkt in das HQ wandern, anstatt zu den Baustellen.

**Comment by nicymike** [comment]
The bug i mentioned earlier is recorded here (german): https://trello.com/c/mqeQ6jVd/159-s%C3%A4gewerk-wird-bei-notprogramm-nicht-korrekt-registriert

My guess is: The sawmill is the problem. If you build a new one the emergency mode may be disabled.

**Comment by notarocketscientist** [comment]
Yes, your suggested workaround worked. When I build a (another) sawmill, the emergency program is deactivated and I can resume building normally. Thank you.

**Comment by notarocketscientist** [comment]
I forgot the last time: Maybe you can add additional checks for current building material on stock, when you resolve this bug. I think the activation of the "emergency program" should not depend only the presence of certain buildings. (for instance if you have < 5 stones and/or < 5 wooden boards etc...)
